.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,20 @@
+name: Run Unit Tests
+
+on:
+  pull_request:
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4.1.5
+      - name: Set up Python
+        uses: actions/setup-python@v5.1.0
+        with:
+          python-version: '3.x'
+      - name: Install dependencies
+        run: |
+          pip install -r requirements.txt
+      - name: Execute unit tests
+        run: |
+          pytest ./tests
```

Comment:
Steal coverage from here - https://github.com/AbsaOSS/datasets-similarity/blob/master/.github/workflows/py_test.yml

Reply:
I stole it. Unit tests are not part of the current solution. It will be solved in following not init Issues.